packages/devtools_app/lib/src/service/service_manager.dart

```diff
@@ -163,6 +163,24 @@ class ServiceConnectionManager {
     return listenable;
   }
 
+  String? _cachedRootLibKey;
+  String? _cachedRootLibValue;
+
+  /// Returns root library or empty string.
+  Future<String?> tryToDetectRootLib() async {
```

Comment:
will we ever try to detect the root library for an isolate that is not the main isolate? Caching the value of the root library and only grabbing it for the main isolate will be an issue if that is the case

Reply:
renamed to clarify that we use main isolate